components/rmrk/Gallery/Holder/Holder.vue

```diff
@@ -0,0 +1,439 @@
+        if (!itemRowMap[nftId]) {
+          itemRowMap[nftId] = {
+            Item: newEvent['nft'],
+            Holder: '-',
+            Bought: 0,
+            Sale: 0,
+            ...commonInfo,
+          }
+        }
+      } else if (newEvent['interaction'] === 'BUY') {
+        const bought = parseInt(newEvent['meta'])
+        if (itemRowMap[nftId]) {
+          if (!('Bought' in itemRowMap[nftId])) {
+            itemRowMap[nftId]['Bought'] = bought
+          }
+        } else {
+          itemRowMap[nftId] = {
+            Item: newEvent['nft'],
+            Holder: newEvent['caller'],
+            Bought: bought,
+            Sale: 0,
+            ...commonInfo,
+          }
+        }
+      }
+    }
+    return Object.values(itemRowMap)
+  }
+
+  private generateHolderGroups(itemRowList): TableRow[] {
+    const holderGroups: Record<string, TableRow> = {}
+    itemRowList
+      .filter((item) => item.Holder !== '-')
+      .forEach((item) => {
+        const holder = item['Holder']
+        if (holderGroups[holder]) {
+          holderGroups[holder].Items.push(item)
+          holderGroups[holder]['Bought'] =
+            holderGroups[holder]['Bought'] + item['Bought']
+          holderGroups[holder]['Sale'] =
+            holderGroups[holder]['Sale'] + item['Sale']
+        } else {
+          holderGroups[holder] = {
+            ...item,
+            Items: [item],
+          }
+        }
+      })
+
+    const holderGroupsList = Object.values(holderGroups)
+
+    holderGroupsList.forEach((group) => {
+      parsePriceForItem(group, this.decimals, this.unit)
+      group['Amount'] = group['Items'].length
+      group['Items'].forEach((item) => {
+        parsePriceForItem(item, this.decimals, this.unit)
+      })
+    })
+    return holderGroupsList
+  }
+
+  protected getBlockUrl(block: string): string {
+    return urlBuilderBlockNumber(
+      block,
+      this.$store.getters['explorer/getCurrentChain'],
+      'subscan'
+    )
+  }
+
+  @Watch('events', { immediate: true })
+  public watchEvent(): void {
+    if (this.events) {
+      this.createTable()
+    }
+  }
+
+  @Watch('currentPage')
+  watchPageValue(val) {
+    this.replaceUrl(String(val))
+  }
+
+  @Debounce(100)
+  replaceUrl(value: string, key = 'page') {
+    this.$router
+      .replace({
+        path: String(this.$route.path),
+        query: { ...this.$route.query, [key]: value },
+      })
+      .catch(console.warn /*Navigation Duplicate err fix later */)
+  }
+}
+</script>
+<style lang="scss">
+.short-identity__table {
+  max-width: 50em;
+  overflow: hidden;
+  text-overflow: ellipsis;
+}
+
+.type-table {
+  white-space: nowrap;
+}
+
+.collapseHidden {
+  .collapse-trigger {
+    display: none;
+  }
+}
+
+.box {
+  .table-nav {
+    display: flex;
+    justify-content: space-between;
+  }
+  .box-container {
+    @media screen and (max-width: 768px) {
+      flex-direction: column-reverse;
+    }
+  }
+}
```

Comment:
this will be global style, could you try to make it scoped?

Reply:
ok